### PR TITLE
Fix edge-case where Texture#toString has a NullPointer because gson s…

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**
…et a null metadata

Also changes NORMAL name to WIDE since that's the accurate name used inside the mc client.

### CODE REVIEW

**Comment by AlexProgrammerDE** [comment]
Mojang AuthLib has similar code:
image

**Comment by AlexProgrammerDE** [comment]
Also fixes now #837

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java [hunk 1]
```diff
@@ -390,7 +389,7 @@ public static class Texture {
          */
         public Texture(String url, Map<String, String> metadata) {
             this.url = url;
-            this.metadata = new HashMap<>(metadata);
+            this.metadata = metadata;
```
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java:392]: Why was this changed? Is the metadata supposed to be mutable now? (Assuming that was the purpose of the copy)

Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java:392]: IntelliJ would claim metadata is never null while it can be null due to gson. Idk how gson does it, but it's able to set the metadata field to null.

Comment (onebeastchris) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java:392]: Interesting - is null metadata valid?

Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java:392]: yeah when a skin is wide, metadata field is missing from textures, if it's slim the metadata field is present

Comment (onebeastchris) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java:392]: <https://github.com/tinytengu/minecraft-authlib/blob/main/src/main/java/com/mojang/authlib/minecraft/MinecraftProfileTexture.java> Got curious and checked, seems to be what minecraft's authlib is doing, so that looks fine.

**Review by onebeastchris — APPROVED** [review]
lgtm
